api/src/db/database.rs

```diff
@@ -4840,6 +4840,40 @@ impl Database {
 
     Ok((total_scopes as usize, scopes))
   }
+
+  pub async fn get_recent_packages_by_user(
+    &self,
+    user_id: &uuid::Uuid,
+  ) -> Result<Vec<Package>> {
+    let packages = sqlx::query_as!(
+        Package,
+        r#"
+        SELECT DISTINCT ON (packages.scope, packages.name)
+          packages.scope as "scope: ScopeName",
+          packages.name as "name: PackageName",
+          packages.description,
+          packages.github_repository_id,
+          packages.runtime_compat as "runtime_compat: RuntimeCompat",
+          packages.when_featured,
+          packages.is_archived,
+          packages.readme_source as "readme_source: ReadmeSource",
+          packages.updated_at,
+          packages.created_at,
+          (SELECT COUNT(created_at) FROM package_versions WHERE scope = packages.scope AND name = packages.name) as "version_count!",
+          (SELECT version FROM package_versions WHERE scope = packages.scope AND name = packages.name AND version NOT LIKE '%-%' AND is_yanked = false ORDER BY version DESC LIMIT 1) as "latest_version"
+        FROM packages
+        JOIN scope_members ON packages.scope = scope_members.scope
+        WHERE scope_members.user_id = $1
```

Comment:
this doesnt seem correct to me; the goal here is any recent publishes by a specific user, but this will list recent publishes for scopes the user is in. this should instead check for publishing_tasks & check its user_id & that its a success

Reply:
hooo got it ! I don't fully understand the codebase yet. Thanks for the input.

Reply:
normally it's better